integration/fuse/src/main/java/alluxio/jnifuse/struct/FileStat.java

```diff
@@ -72,13 +72,13 @@ public static boolean S_ISLNK(int mode) {
   public FileStat(ByteBuffer buffer) {
     super(buffer);
     if (OSUtils.isMacOS()) {
-      st_dev = new Unsigned64();
-      st_mode = new Unsigned32();
-      st_nlink = new Unsigned64();
+      st_dev = new Unsigned32();
```

Comment:
How do we know what the types are supposed to be?

Reply:
basically I checked out the src code of corresponding definition in MacOS and Linux in their kernels.
It looks like Linux and MacOS are using different data types for these structures.
We may need an indirection here.